### PR TITLE
add support for feature flags

### PR DESCRIPTION
**Pull request by Dotunj**

### CODE REVIEW

**Review by jirevwe — APPROVED** [review]

File: internal/pkg/fflag/flipt/client.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
+package flipt
```
Comment (Subomi) [inline internal/pkg/fflag/flipt/client.go:1]: @Dotunj This library isn't doing too much; we shouldn't be logging here. The client code should handle logging while this library returns errors.

File: cmd/main.go [hunk 2]
```diff
@@ -112,6 +114,7 @@ type app struct {
 	cache    cache.Cache
 	limiter  limiter.RateLimiter
 	searcher searcher.Searcher
+	fflag    fflag.FeatureFlag
```
Comment (Subomi) [inline cmd/main.go:117]: @Dotunj We don't need this DI from the `cmd` package. From my understanding, this is only an HTTP Client. We should instantiate it where it's needed and call the API.

File: internal/pkg/fflag/noop/client.go [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package noopfflag
```
Comment (Subomi) [inline internal/pkg/fflag/noop/client.go:1]: @Dotunj I don't think we need a noop for this. From my understanding, It's an HTTP wrapper.

**Review by Subomi — CHANGES_REQUESTED** [review]
Nice! Please see below:

**Review by jirevwe — APPROVED** [review]

**Review by danvixent — APPROVED** [review]

**Pull request merged**